src/cockpit/channels/filesystem.py

```diff
@@ -503,9 +503,10 @@ def do_open(self, options: JsonObject) -> None:
         if not os.path.isabs(self.path):
             raise JsonError(options, '"path" must be an absolute path')
 
-        self.getattrs = self.make_getattrs(get_strv(options, 'attrs'))
-        self.fnmatch = get_str(options, 'fnmatch', '')
-        self.targets = get_str(options, 'targets', '') == 'stat'
+        attrs = set(get_strv(options, 'attrs'))
+        self.getattrs = self.make_getattrs(attrs - {'targets', 'entries'})
```

Comment:
Ah, this is a bit brittle. Could make_getattrs change this:
```py
 stat_getters = tuple((key, available_stat_getters.get(key, lambda _: None)) for key in attrs)
```

to use `available_stat_getters[key]` without the lambda: none fallback? That would make it more robust and instantly tell us instead of silently ignoring unknown attributes.

Reply:
The `None` fallback is important there for (for example) ensuring that an attribute like 'target' gets unset if it's not specifically set in an iteration.  The general idea here, as well, is to silently ignore unknown attributes in an attempt to remain forward-compatible.

Reply:
Ah, ok. So we need to be really careful about missing stat-ers and other attributes in `attrs`

Reply:
I mean, the semantics here are "if there's a problem reporting that attribute, then it's not reported".  An attribute being unsupported in a particular bridge version (or you making a typo) would indeed count as a problem. :)